src/storage/txn/actions/prewrite.rs

```diff
@@ -412,6 +438,98 @@ impl<'a> PrewriteMutation<'a> {
         .into())
     }
 
+    fn check_assertion<S: Snapshot>(
+        &self,
+        reader: &mut SnapshotReader<S>,
+        write: &Option<(Write, TimeStamp)>,
+        write_loaded: bool,
+    ) -> Result<(Option<(Write, TimeStamp)>, bool)> {
+        if self.assertion == Assertion::None
+            || self.txn_props.assertion_level == AssertionLevel::Off
+        {
+            MVCC_PREWRITE_ASSERTION_PERF_COUNTER_VEC.none.inc();
+            return Ok((None, false));
+        }
+
+        if self.txn_props.assertion_level != AssertionLevel::Strict && !write_loaded {
+            MVCC_PREWRITE_ASSERTION_PERF_COUNTER_VEC
+                .write_not_loaded_skip
+                .inc();
+            return Ok((None, false));
+        }
+
+        let mut reloaded_write = None;
+        let mut reloaded = false;
+
+        // To pass the compiler's lifetime check.
+        let mut write = write;
+
+        if write_loaded
+            && write.as_ref().map_or(
+                false,
+                |(w, _)| matches!(w.gc_fence, Some(gc_fence_ts) if !gc_fence_ts.is_zero()),
+            )
+        {
+            // The previously-loaded write record has an invalid gc_fence. Regard it as none.
+            write = &None;
+        }
+
+        // Load the most recent version if prev write is not loaded yet, or the prev write is not
+        // a data version (`Put` or `Delete`)
+        let need_reload = !write_loaded
+            || write.as_ref().map_or(false, |(w, _)| {
```

Comment:
Is it possible to reload if the level is not `STRICT`?

Reply:
Yes but only when it has already loaded a write but it's a Rollback or a Lock. In this case reload might happen, but it's expected to be relatively cheap since it only need to do `next`s, so I think that should be ok